### PR TITLE
create custom sap-btp-operator helm chart to allow various cert methods

### PR DESCRIPTION
**Pull request by voigt**
This PR adds a custom helm chart thats allows for a more flexible sap-btp-operator deployments using custom certificates.

This enables deployments with:

- custom self-signed CA
- jetstack/cert-manager
- gardener/cert-management (either self-signed or defined issuer)
- per default the sap-btp-operator now uses a hard coded, self-signed CA. This removes the hard requirement of having  jetstack/cert-manager running in the cluster

### CODE REVIEW

File: chart/templates/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  namespace: {{.Release.Namespace}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      control-plane: controller-manager
+  template:
+    metadata:
+      annotations:
+        rollme: {{ randAlphaNum 5 | quote }}
```
Comment (wozniakjan) [inline chart/templates/deployment.yml:16]: just out of curiosity, do you know what this is for?

File: chart/templates/secret-certificate.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
...0hycXJoQ1VGQ2d2ME9tKzhFVGVMaFNUalE3YjY4MzRLbUc5UkYwWERZdApiWE1jZThrM2IvNGtuRkhIblNiWUhmRjlGK2llNlhZV2w2NEtYN0lQT0w5K1l3ZFdzQ1JESitpZ3E2eG1kbm8yCktvUEUzTVFFMGoxMy9ZR1dzOGNDQXdFQUFhT0NBUWt3Z2dFRk1BNEdBMVVkRHdFQi93UUVBd0lGb0RBZEJnTlYKSFNVRUZqQVVCZ2dyQmdFRkJRY0RBUVlJS3dZQkJRVUhBd0l3REFZRFZSMFRBUUgvQkFJd0FEQWRCZ05WSFE0RQpGZ1FVbnhyNllNbEhmcVJ0YTBwYnkzYktlbWFzQ3U0d0h3WURWUjBqQkJnd0ZvQVVnb1RiSlUycVFoM3F4Rm5VCmk0ZVRTeEEyTmJzd2dZVUdBMVVkRVFSK01IeUNOWE5oY0MxaWRIQXRiM0JsY21GMGIzSXRkMlZpYUc5dmF5MXoKWlhKMmFXTmxMbk5oY0MxaWRIQXRiM0JsY21GMGIzSXVjM1pqZ2tOellYQXRZblJ3TFc5d1pYSmhkRzl5TFhkbApZbWh2YjJzdGMyVnlkbWxqWlM1ellYQXRZblJ3TFc5d1pYSmhkRzl5TG5OMll5NWpiSFZ6ZEdWeUxteHZZMkZzCk1BMEdDU3FHU0liM0RRRUJDd1VBQTRJQkFRQzQzVXl0RDFFdzNzdnZwYUc1SEh3OXp4ZDZ4YUhYQytYZnBWanQKbmt4WUV1Z3pzRzk4Ylk2eEIvRnBGbHJxdENMSVNmSGo0cnFmZUQzRGhBRjArbmdxYllPQkxQWW16SmJNVmxqbwpZQmpJMW5HdmRKSUpBazhWVU1GM2ZBbjlGWjQyV3k0MmtGeDNyM21WQUtWOVNFNWVCdUVWazllaW9IRjI5LzMwClg0bFhObWp1b0tNcTM2ZkxzQmRCL0FhUTFEWVpJS0NvcDM4bTY4M3pxejA0cGMvMkMyczJmbHMrYWRiSEtIOHYKSUFxVU83YnZ2eFZjMi9jb1JoWkZ2c29kYnd0YXUvNVRzbnNDUnI4d0dRM1V6bCtDNUc0Z05MMEkrNldrbkV2YgpMbkJoaU42SHl6TjdXYk9DUE0zUE9GTnpkWEFvb2Q1VmpCLy9RM0k0RkpzNmlNeUsKLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQo="
```
Comment (wozniakjan) [inline chart/templates/secret-certificate.yml:20]: nit: this expires in `Not After : Jun 25 16:06:00 2022 GMT`, we should set a reminder to update this before then. Also, given these are meant as a convenience for testing, I'd be ok setting the expiration date to something like 10y in future. What do you think?

**Review by wozniakjan — APPROVED** [review]
lgtm, great job!

**Comment by wozniakjan** [comment]
btw. I just got to testing it and either I'm missing something or there might be some bug with the default option hardcoded certs
```
$ pwd
.../go/src/github.com/SAP/sap-btp-service-operator/chart

$ git log -1 --oneline
9a43d38 (HEAD -> add_additional_cert_options, voigt/add_additional_cert_options) create custom sap-btp-operator helm chart to allow various cert methods

$ helm install sap-btp-operator ./ --atomic --create-namespace --namespace='sap-btp-operator' --set manager.secret.clientid="mock" --set manager.secret.clientsecret="mock" --set manager.secret.url="mock" --set manager.secret.tokenurl="mock" --set cluster.id="mock"
Error: template: sap-btp-operator/templates/cert-management-issuer.yml:1:18: executing "sap-btp-operator/templates/cert-management-issuer.yml" at <.Values.manager.certificates.certManagement.issuer>: nil pointer evaluating interface {}.issuer
```

File: chart/templates/deployment.yml (hunk 1 above)
Comment (voigt) [reply chart/templates/deployment.yml:16]: I think [thats why](https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments).

File: chart/templates/secret-certificate.yml (hunk 2 above)
Comment (voigt) [reply chart/templates/secret-certificate.yml:20]: Totally agree. I will adjust it accordingly - good point!

**Comment by voigt** [comment]
> btw. I just got to testing it and either I'm missing something or there might be some bug with the default option hardcoded certs

Yike! Sorry to waste your time - fixed the issue in [667283e](https://github.com/kyma-incubator/sap-btp-service-operator/pull/1/commits/667283e1526196ca45218ea1889b1a405788e9e9).

**Review by wozniakjan — APPROVED** [review]
lgtm,

I think you will need to sign the license/cla in order to be able to merge the PR

**Pull request merged**